File: javascript/src/problems/best-time-to-buy-and-sell-stock.js
/**
 * 121. Best Time to Buy and Sell Stock [easy] https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
 * 
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let min = prices[0];
    let max = -1;
    let income = 0;
    for (let i = 1; i < prices.length; i++) {
        if (prices[i] < min) {
            min = prices[i];
            max = -1;
        } else if (prices[i] > max) {
            max = prices[i];
            income = Math.max(income, max - min);
        }
    }
    return income;
};

module.exports = { maxProfit };

File: java/src/main/java/org/carlosbello/leetcode/problems/BestTimeToBuyAndSellStock.java
package org.carlosbello.leetcode.problems;

/**
 * 121. Best Time to Buy and Sell Stock [easy] https://leetcode.com/problems/best-time-to-buy-and-sell-stock/
 */
public class BestTimeToBuyAndSellStock {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) return 0;

        int max = 0;
        int bestBy = prices[0];
        for (int i = 1; i < prices.length; i++) {
            if (prices[i] < bestBy) {
                bestBy = prices[i];
            } else if (prices[i] - bestBy > max) {
                max = prices[i] - bestBy;
            }
        }
        return max;
    }
}
